pipelines/simple-notebook-pipeline/Simple Notebook Pipeline.ipynb

```diff
@@ -0,0 +1,285 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# If you've already done his during setup you don't need to do it again\n",
+    "KFP_PACKAGE = 'https://storage.googleapis.com/ml-pipeline/release/0.1.23/kfp.tar.gz'\n",
+    "!pip3 install $KFP_PACKAGE --upgrade"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setup\n",
+    "<b> Important: </b> Make sure to use your GCP project name and a GCS bucket for saving the pipelines.  If you haven't already, [setup a GCS bucket](https://cloud.google.com/storage/docs/creating-buckets)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Set your output directory and project name. \n",
+    "PROJECT_NAME =  '[YOUR-GCP-PROJECT-NAME]'           # 'Your-Gcp-Project-Name'\n",
+    "OUTPUT_DIR = 'gs://[YOUR-GCS-BUCKET]/assets'        # A path for asset outputs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "EXPERIMENT_NAME = 'Simple notebook pipeline'                    # Name of the experiment in the UI\n",
+    "BASE_IMAGE = 'tensorflow/tensorflow:2.0.0b0-py3'    # Base image used for components in the pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp\n",
+    "import kfp.dsl as dsl\n",
+    "from kfp import compiler"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create pipeline component"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create python function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dsl.python_component(\n",
+    "    name='add_op',\n",
+    "    description='adds two numbers',\n",
+    "    base_image=BASE_IMAGE  # you can define the base image here, or when you build in the next step. \n",
+    ")\n",
+    "def add(a: float, b: float) -> float:\n",
+    "    '''Calculates sum of two arguments'''\n",
+    "    print(a, '+', b, '=', a + b)\n",
+    "    return a + b"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Build pipeline component from the function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Docker image for the pipeline component\n",
+    "TARGET_IMAGE = 'gcr.io/%s/add-op:latest' % PROJECT_NAME \n",
+    "\n",
+    "# The return value \"DeployerOp\" represents a step that is used in a pipeline\n",
+    "add_op = compiler.build_python_component(\n",
```

Comment:
When executing this cell, I'm getting a permissions error:
`Forbidden: 403 GET https://www.googleapis.com/storage/v1/b/kubeflow-examples-ner?projection=noAcl: kubeflow-vm@mcas-195423.iam.gserviceaccount.com does not have storage.buckets.get access to kubeflow-examples-ner.`
This is a new bucket that I created with this command:
`gsutil mb gs://kubeflow-examples-ner`

Check with the pipelines team for their recommendation here. Does it need to be a public bucket, do we need to add permissions for this service account explicitly, either at the role or bucket level, or should we authenticate explicitly in a different way?

Reply:
It may be an issue with my environment, so if you can verify that someone else is able to run this from scratch using these instructions, that's good enough for me.

Reply:
So the reason you are getting the permissions error is because the bucket you created is not associated with the GCP project you are running kubeflow with.  I'll specify in the instructions when creating the GCP Bucket to use  `-p <project_name>`  if you use `gsutil` or to create the bucket in the same project if you use the UI.